aarch64_extension.cpp

```diff
@@ -369,20 +491,29 @@ class AArch64ArchitectureExtension : public ArchitectureHook {
           1 == detail->operands[1].vector_index && 8 == Rd_size &&
           16 == Rn_size) {
         /* FMOV <Xn>, <Vd>.D[1] */
-        il.AddInstruction(il.SetRegister(
-            Rd_size, Rd,
-            il.LowPart(Rd_size,
-                       il.LogicalShiftRight(Rn_size, il.Register(Rn_size, Rn),
-                                            il.Const(1, 64)))));
+        uint32_t subRn = getVectorSubregister(Rn, Rd_size, 1);
+        if (0 == subRn) {
+          LogError("%#lx: Invalid vector element access in operand 1",
+                   instr->address);
+          return false;
+        }
+
+        il.AddInstruction(
+            il.SetRegister(Rd_size, Rd, il.Register(Rd_size, subRn)));
         return true;
       } else if (1 == detail->operands[0].vector_index &&
                  -1 == detail->operands[1].vector_index && 16 == Rd_size &&
                  8 == Rn_size) {
         /* FMOV <Vd>.D[1], <Xn> */
-        il.AddInstruction(il.SetRegister(
-            Rd_size, Rd,
-            il.ZeroExtend(Rd_size, il.Or(Rn_size, il.Register(Rn_size, Rn),
-                                         il.Register(Rn_size, Rd)))));
+        uint32_t subRd = getVectorSubregister(Rd, Rn_size, 1);
+        if (0 == subRd) {
+          LogError("%#lx: Invalid vector element access in operand 0",
```

Comment:
Looks like a typo, did you mean `#%lx`?

Reply:
`%#lx` => `%`:format `#`:alternate `l`:long `x`:hex
The alternate modifier adds a `0x` prefix for us.

Reply:
Very cool, I did not know that. TIL.